test/test_common/network_utility.cc

```diff
@@ -116,10 +116,25 @@ Address::InstanceConstSharedPtr getCanonicalLoopbackAddress(Address::IpVersion v
   return Network::Utility::getIpv6LoopbackAddress();
 }
 
-Address::InstanceConstSharedPtr getAnyAddress(const Address::IpVersion version) {
+// There is no portable way to initialize sockaddr_in6 with a static initializer, do it with a
+// helper function instead
+static sockaddr_in6& v6any_() {
+  static sockaddr_in6 v6any = {};
+  v6any.sin6_family = AF_INET6;
+  v6any.sin6_addr = in6addr_any;
+
+  return v6any;
+}
+
+Address::InstanceConstSharedPtr getAnyAddress(const Address::IpVersion version, bool v4_compat) {
   if (version == Address::IpVersion::v4) {
     return Network::Utility::getIpv4AnyAddress();
   }
+  if (v4_compat) {
+    static sockaddr_in6 v6any = v6any_();
+    static Address::InstanceConstSharedPtr any(new Address::Ipv6Instance(v6any, false));
```

Comment:
I would just call v6any_() on this line instead of initializing an extra static var. TIOLI.

Reply:
Done, thanks!